packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -132,6 +133,24 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
     return { ...this.defaultBIP44Params, ...params }
   }
 
+  getBIP44Params(params: GetBIP44ParamsInput): BIP44Params {
+    let purpose: number
+    switch (params.accountType) {
+      case UtxoAccountType.SegwitNative:
+        purpose = 84
+        break
+      case UtxoAccountType.SegwitP2sh:
+        purpose = 49
+        break
+      case UtxoAccountType.P2pkh:
+        purpose = 44
+        break
+      default:
+        throw new Error(`not a supported accountType ${params.accountType}`)
+    }
```

Comment:
IIFE pattern to avoid mutation 🙏 

```suggestion
    const purpose: number = (() => {
      switch (params.accountType) {
        case UtxoAccountType.SegwitNative:
          return 84
        case UtxoAccountType.SegwitP2sh:
          return 49
        case UtxoAccountType.P2pkh:
          return 44
        default:
          throw new Error(`not a supported accountType ${params.accountType}`)
      }
    })()
```

Reply:
very nice ty